src/adapter/edges.rs

```diff
@@ -261,6 +262,35 @@ pub(super) fn resolve_function_like_edge<'a, V: AsVertex<Vertex<'a>> + 'a>(
     }
 }
 
+pub(super) fn resolve_method_receiver_edge<'a, V: AsVertex<Vertex<'a>> + 'a>(
+    contexts: ContextIterator<'a, V>,
+    edge_name: &str,
+) -> ContextOutcomeIterator<'a, V, VertexIterator<'a, Vertex<'a>>> {
+    match edge_name {
+        "receiver" => resolve_neighbors_with(contexts, move |vertex| {
+            let origin = vertex.origin;
+            let method = vertex.as_function().expect("vertex was not a Function");
+
+            // Check if the first parameter is a self receiver
+            let receiver = method.sig.inputs.first().and_then(|(name, ty)| {
+                if name.starts_with("self") {
+                    Some(MethodSelfReceiver::new(ty))
+                } else {
+                    None
+                }
+            });
+
+            match receiver {
+                Some(receiver) => Box::new(std::iter::once(
+                    origin.make_method_receiver_vertex(receiver),
+                )),
+                None => Box::new(std::iter::empty()),
+            }
```

Comment:
There's a cute trick you can do here: `Option<T>` has [`.into_iter()`](https://doc.rust-lang.org/std/option/enum.Option.html#method.into_iter) so you can do:
```suggestion
            Box::new(receiver.into_iter().map(|r| origin.make_method_receiver_vertex(r)))
```